diff --git a/usr/share/lipstick-jolla-home-qt5/layers/EdgeLayer.qml b/usr/share/lipstick-jolla-home-qt5/layers/EdgeLayer.qml
index 26e0dfd..003c32b 100644
--- a/usr/share/lipstick-jolla-home-qt5/layers/EdgeLayer.qml
+++ b/usr/share/lipstick-jolla-home-qt5/layers/EdgeLayer.qml
@@ -178,13 +178,13 @@ Layer {
             NumberAnimation {
                 target: edgeLayer
                 property: "x"
-                duration: 300
+                duration: 250
                 easing.type: Easing.OutQuad
             }
             NumberAnimation {
                 target: edgeLayer
                 property: "y"
-                duration: 300
+                duration: 250
                 easing.type: Easing.OutQuad
             }
             PropertyAction {
@@ -199,13 +199,13 @@ Layer {
             NumberAnimation {
                 target: edgeLayer
                 property: "x"
-                duration: 400
+                duration: 350
                 easing.type: Easing.InOutQuad
             }
             NumberAnimation {
                 target: edgeLayer
                 property: "y"
-                duration: 400
+                duration: 350
                 easing.type: Easing.InOutQuad
             }
             PropertyAction {
@@ -224,7 +224,7 @@ Layer {
             id: hintShowAnimation
             target: edgeLayer
             property: "_hintProgress"
-            duration: 300
+            duration: 250
             to: Theme.iconSizeLauncher * 2
             easing.type: Easing.InOutQuad
         }
@@ -232,7 +232,7 @@ Layer {
         NumberAnimation {
             target: edgeLayer
             property: "_hintProgress"
-            duration: 300
+            duration: 250
             to: 0
             easing.type: Easing.InOutQuad
         }
diff --git a/usr/share/lipstick-jolla-home-qt5/layers/HomeLayer.qml b/usr/share/lipstick-jolla-home-qt5/layers/HomeLayer.qml
index 90a20b5..cae53dc 100644
--- a/usr/share/lipstick-jolla-home-qt5/layers/HomeLayer.qml
+++ b/usr/share/lipstick-jolla-home-qt5/layers/HomeLayer.qml
@@ -200,18 +200,6 @@ Pannable {
         SwitcherLayer {
             id: switcherLayer
 
-            readonly property real inactiveScale: Screen.sizeCategory >= Screen.Large ? 0.90 : 0.83
-            readonly property bool moving: homescreen.moving
-                        || (Desktop.instance && Desktop.instance.switcher.moving)
-            property bool inhibitScale
-            property real inhibitedScale
-            onMovingChanged: {
-                if (moving) {
-                    inhibitedScale = scale
-                }
-                inhibitScale = moving
-            }
-
             objectName: "switcherLayer"
 
             leftItem:  eventsLayer
@@ -220,25 +208,7 @@ Pannable {
             width: homescreen.width
             height: homescreen.height
 
-            scale: {
-                if (inhibitScale) {
-                    return inhibitedScale
-                } else if (Desktop.instance && Desktop.instance.switcher.appShowInProgress) {
-                    return inactiveScale
-                } else if (Lipstick.compositor.launcherLayer.exposed) {
-                    return 1 - ((1 - inactiveScale) * Lipstick.compositor.launcherLayer.exposure)
-                } else if (Lipstick.compositor.powerKeyLayer.exposed) {
-                    return 1 - ((1 - inactiveScale) * Lipstick.compositor.powerKeyLayer.exposure)
-                } else if (homescreen.active || Lipstick.compositor.topmostIsDialog) {
-                    return 1
-                } else {
-                    return inactiveScale
-                }
-            }
-            Behavior on scale {
-                enabled: switcherLayer.visible
-                NumberAnimation { duration: 250; easing.type: Easing.OutQuad }
-            }
+            scale: 1
 
             anchors { rightMargin: -rightItem.minimizeMargin }
         }
diff --git a/usr/share/lipstick-jolla-home-qt5/layers/NotificationOverviewLayer.qml b/usr/share/lipstick-jolla-home-qt5/layers/NotificationOverviewLayer.qml
index 9227f3f..e3e2123 100644
--- a/usr/share/lipstick-jolla-home-qt5/layers/NotificationOverviewLayer.qml
+++ b/usr/share/lipstick-jolla-home-qt5/layers/NotificationOverviewLayer.qml
@@ -24,11 +24,6 @@ Layer {
 
     property bool lockScreenLocked
 
-    property real _launcherPosition: Math.max(Lipstick.compositor.launcherLayer.exposure, Lipstick.compositor.powerKeyLayer.exposure)
-    Behavior on _launcherPosition {
-        NumberAnimation { duration: 400; easing.type: Easing.OutQuad }
-    }
-
     exposed: {
         if (!hasNotifications) {
             return false
@@ -44,9 +39,7 @@ Layer {
 
     // Behaviors must be enabled or disabled before dependent values are modified
     property bool _positionBehaviorEnabled: {
-        if (state == "peeking_through_launcher" ||
-            state == "switcher-menu" ||
-            state == "peeking_at_switcher" ||
+        if (state == "peeking_at_switcher" ||
             state == "lockscreen_animated" ||
             state == "lockscreen_unanimated" ||
             state == "peeking_at_eventsview") {
@@ -56,9 +49,7 @@ Layer {
     }
 
     property bool _opacityBehaviorEnabled: {
-        if (state == "peeking_through_launcher" ||
-            state == "switcher-menu" ||
-            state == "peeking_at_switcher" ||
+        if (state == "peeking_at_switcher" ||
             state == "lockscreen_unanimated" ||
             state == "animate_to_eventsview" ||
             state == "peeking_at_eventsview") {
@@ -83,9 +74,7 @@ Layer {
             _state == "lockscreen_unanimated") {
             return Lipstick.compositor.lockScreenLayer.notificationOpacity
         }
-        if (_state == "peeking_through_launcher" ||
-            _state == "switcher-menu" ||
-            _state == "peeking_at_switcher" ||
+        if (_state == "peeking_at_switcher" ||
             _state == "animate_to_eventsview") {
             return 1
         }
@@ -93,11 +82,7 @@ Layer {
     }
 
     property real _x: {
-        if (_state == "switcher-menu") {
-            return notifications.visiblePosition - ((1 - root._launcherPosition) * notifications.margin)
-        }
-        if (_state == "peeking_through_launcher" ||
-            _state == "peeking_at_switcher" ||
+        if (_state == "peeking_at_switcher" ||
             _state == "lockscreen_animated" ||
             _state == "lockscreen_unanimated") {
             return notifications.visiblePosition
@@ -112,23 +97,9 @@ Layer {
 
     states: [
         State {
-            name: "peeking_through_launcher"
-            when: Lipstick.compositor.switcherLayer.isCurrentItem
-                    && Lipstick.compositor.launcherLayer.exposed
-                    && Lipstick.compositor.homePeeking
-                    && !lipstickSettings.lockscreenVisible
-        },
-        State {
-            name: "switcher-menu"
-            when: Lipstick.compositor.switcherLayer.isCurrentItem
-                    && (Lipstick.compositor.launcherLayer.exposed || Lipstick.compositor.powerKeyLayer.exposed)
-                    && !lipstickSettings.lockscreenVisible
-        },
-        State {
             name: "peeking_at_switcher"
             when: Lipstick.compositor.homeLayer.currentItem == Lipstick.compositor.switcherLayer
-                  && ((Lipstick.compositor.peekingLayer.exposed && Lipstick.compositor.peekingAtHome && !lockScreenLocked)
-                        || (Lipstick.compositor.launcherPeeking && !Lipstick.compositor.homeLayer.active && !lockScreenLocked))
+                  && (Lipstick.compositor.peekingLayer.exposed && Lipstick.compositor.peekingAtHome && !lockScreenLocked)
         },
         State {
             // LockScreenLayer can be locked below the running application. Notification opacity and position
diff --git a/usr/share/lipstick-jolla-home-qt5/layers/StackLayer.qml b/usr/share/lipstick-jolla-home-qt5/layers/StackLayer.qml
index 8f8aec5..042fa22 100644
--- a/usr/share/lipstick-jolla-home-qt5/layers/StackLayer.qml
+++ b/usr/share/lipstick-jolla-home-qt5/layers/StackLayer.qml
@@ -329,11 +329,11 @@ Layer {
         // Allow some time for the application to respond to the visibility change before
         // capturing the fade in snapshot.  This is the same amount of time it takes the
         // switcher scale animation to complete so the animations will appear sequential.
-        PauseAnimation { id: delayAnimation; duration: 250 }
+        PauseAnimation { duration: showAnimation.retries == 0 ? 20 : (layer.window.mapped ? 0 : 1000) }
         ScriptAction {
             script: {
                 if (active) {
-                    if (layer.window.mapped || ++showAnimation.retries >= 4) {
+                    if (layer.window.mapped || ++showAnimation.retries > 1) {
                         showAnimation.retries = 0
                         if (layer.mergeWindows) {
                             layer.cacheWindow(layer.window)
@@ -352,7 +352,7 @@ Layer {
         }
         FadeAnimator {
             target: layer
-            duration: 400
+            duration: 350
             from: 0.0
             to: 1.0
             easing.type: Easing.InOutQuad
@@ -363,7 +363,8 @@ Layer {
         target: layer.window
         onMappedChanged: {
             if (layer.window.mapped && showAnimation.retries > 0) {
-                delayAnimation.complete()
+                showAnimation.retries = 0
+                showAnimation.restart()
             }
         }
     }
diff --git a/usr/share/lipstick-jolla-home-qt5/lockscreen/LockScreen.qml b/usr/share/lipstick-jolla-home-qt5/lockscreen/LockScreen.qml
index dc90755..2125026 100644
--- a/usr/share/lipstick-jolla-home-qt5/lockscreen/LockScreen.qml
+++ b/usr/share/lipstick-jolla-home-qt5/lockscreen/LockScreen.qml
@@ -319,12 +319,8 @@ ApplicationWindow {
                             && Lipstick.compositor.systemInitComplete) {
                         lockItem.reset()
                         deviceLockContainer.opacity = 0.0
-                        if (Lipstick.compositor.notificationOverviewLayer.hasNotifications) {
-                            Lipstick.compositor.lockScreenLayer.notificationAnimation = "animated"
-                        } else {
-                            Lipstick.compositor.lockScreenLayer.notificationAnimation = "immediate"
-                            Lipstick.compositor.goToEvents()
-                        }
+                        Lipstick.compositor.lockScreenLayer.notificationAnimation = "immediate"
+                        Lipstick.compositor.goToEvents()
                     } else {
                         lockItem.reset()
                         Lipstick.compositor.goToSwitcher(false)
diff --git a/usr/share/lipstick-jolla-home-qt5/main/PeekArea.qml b/usr/share/lipstick-jolla-home-qt5/main/PeekArea.qml
index 265e1ab..1ace827 100644
--- a/usr/share/lipstick-jolla-home-qt5/main/PeekArea.qml
+++ b/usr/share/lipstick-jolla-home-qt5/main/PeekArea.qml
@@ -79,7 +79,7 @@ Item {
     Behavior on opacity {
         id: opacityBehavior
         enabled: false
-        SmoothedAnimation { id: opacityAnimation; duration: 400; velocity: 1000 / duration }
+        SmoothedAnimation { id: opacityAnimation; duration: 300; velocity: 1000 / duration }
     }
 
     Item {
@@ -127,7 +127,7 @@ Item {
             opacityBehavior.enabled = false
             if (peekFilter.closing) {
                 peekArea._closeAnimating = true
-                clipEndAnimation.duration = 400
+                clipEndAnimation.duration = 300
             } else {
                 clipEndAnimation.duration = 200
                 clipEndAnimation.to = 0
diff --git a/usr/share/lipstick-jolla-home-qt5/switcher/SwitcherItem.qml b/usr/share/lipstick-jolla-home-qt5/switcher/SwitcherItem.qml
index b6c281a..2d47e5d 100644
--- a/usr/share/lipstick-jolla-home-qt5/switcher/SwitcherItem.qml
+++ b/usr/share/lipstick-jolla-home-qt5/switcher/SwitcherItem.qml
@@ -119,26 +119,6 @@ EditableGridDelegate {
         }
     }
 
-    Connections {
-        target: wrapper.currentApp ? Lipstick.compositor : null
-        ignoreUnknownSignals: true
-        onHomePeekingChanged: {
-            if (Lipstick.compositor.homePeeking && !wrapper.closing) {
-                contentItem.opacity = 0.0
-                homePeekAnimation.restart()
-            }
-        }
-    }
-    SequentialAnimation {
-        id: homePeekAnimation
-        PauseAnimation { duration: 200 }
-        FadeAnimation {
-            target: contentItem
-            duration: 400
-            to: 1.0
-        }
-    }
-
     property bool hideCover: wrapper.launching || startupTimer.running || !windowPixmap.hasPixmap
     property real coverOpacity: hideCover ? 0.0 : (!running ? 0.6 : 1.0)
 
@@ -187,7 +167,7 @@ EditableGridDelegate {
         }
 
         Behavior on opacity {
-            FadeAnimation { id: coverOpacityAnimation; duration: 500 }
+            FadeAnimation { id: coverOpacityAnimation; duration: 350 }
         }
     }
 
@@ -198,7 +178,7 @@ EditableGridDelegate {
         icon: launcherItem ? launcherItem.iconId : ""
         layer.effect: null
         opacity: hideCover ? 1.0 : 0.0
-        Behavior on opacity { FadeAnimation { duration: 500 } }
+        Behavior on opacity { FadeAnimation { duration: 350 } }
     }
 
     Loader {
@@ -227,7 +207,7 @@ EditableGridDelegate {
     Rectangle {
         color: Theme.highlightBackgroundColor
         anchors.fill: parent
-        opacity: ((currentApp && !Lipstick.compositor.homePeeking) || windowMapPending || launching || wrapper.closing) ? 0.4 : (contentItem.pressed ? Theme.highlightBackgroundOpacity : 0)
+        opacity: (currentApp || windowMapPending || (launching && !minimized) || wrapper.closing) ? 0.4 : (contentItem.pressed ? Theme.highlightBackgroundOpacity : 0)
         radius: windowPixmap.radius
         Behavior on opacity {
             enabled: !contentItem.pressed
